nix/default.nix

```diff
@@ -0,0 +1,271 @@
+        "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBHnjecqMe2lrGzAvQ2VQRTXhjZ5q1tONgme+2/97Z3VSXdY0i2bEH3qGEIC7uMyWUfmLystXxqP0u6/Xspmm0Ck="
+      ];
+    in
+    {
+      config,
+      ...
+    }:
+    {
+
+      services.tailscale.enable = true;
+      services.tailscale.authKeyFile = config.sops.secrets.tailscale_auth_key.path;
+      services.tailscale.extraUpFlags = [ "--ssh" ];
+
+      users.users.root.openssh.authorizedKeys.keys = keys;
+      users.users.komunix = {
+        home = "/home/komunix";
+        createHome = true;
+        isNormalUser = true;
+        extraGroups = [
+          "wheel"
+          "networkmanager"
+        ];
+        openssh.authorizedKeys.keys = keys;
+      };
+
+      imports = [
+        inputs.sops.nixosModules.sops
+      ];
+
+      sops.defaultSopsFile = ../secrets/secret.yaml;
+      sops.age.sshKeyPaths = [ "/etc/ssh/ssh_host_ed25519_key" ];
+      sops.age.keyFile = "/var/lib/sops-nix/key.txt";
+      sops.age.generateKey = true;
+      sops.secrets.tailscale_auth_key = { };
+    };
+
+  flake.nixosModules.services-cachex =
+    {
+      pkgs,
+      config,
+      lib,
+      ...
+    }:
+    let
+      cfg = config.services.cachex;
+    in
+    with lib;
+    {
+      options.services.cachex = {
+        enable = mkOption {
+          default = false;
+          type = with types; bool;
+          description = ''
+            Enable caddy for cachex
+          '';
+        };
+        cachexPackage = mkOption {
+          type = types.package;
+          description = ''
+            Cachex Package
+          '';
+        };
+        caddyPackage = mkOption {
+          default = pkgs.caddy;
+          type = types.package;
+          description = ''
+            Caddy package
+          '';
+          example = literalExample "pkgs.caddy";
+        };
+        settings.cron = mkOption {
+          default = false;
+          type = with types; bool;
+          description = ''
+            Enable cachex in cron job
+          '';
+        };
+        settings.workDir = mkOption {
+          type = types.str;
+          example = literalExample "/home/komunix";
+        };
+        settings.listenAddress = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            Listen Address
+          '';
+        };
+        settings.listenPort = mkOption {
+          type = types.port;
+          default = 2022;
+          description = ''
+            Listen Address
+          '';
+        };
+      };
+
+      config = mkIf cfg.enable {
+        environment.systemPackages = [ cfg.caddyPackage ];
+
+        services.cron.enable = cfg.settings.cron;
+        services.cron.systemCronJobs = [
+          "* * * * * cachex ${getExe cfg.cachexPackage} > ${cfg.settings.workDir}/cachex/index.html"
+        ];
+
+        services.rpcbind.enable = true; # needed for NFS
+        systemd.mounts = [
```

Comment:
Please review this @faultables 

in local vm of komunix, have `/home/komunix/nfs` but shown error "no devices"

Reply:
Current progress (Error) @faultables 

```bash
home-komunix-nfs.mount - /home/komunix/nfs
     Loaded: loaded (/etc/systemd/system/home-komunix-nfs.mount; linked; preset: enabled)
     Active: failed (Result: exit-code) since Sun 2024-11-10 13:43:25 UTC; 29s ago
 Invocation: 0e6ad806b6eb4acd8a0423850f82379b
TriggeredBy: ● home-komunix-nfs.automount
      Where: /home/komunix/nfs
       What: synology:/volume2/komunix
         IP: 984B in, 1.4K out
   Mem peak: 1.5M
        CPU: 60ms

Nov 10 13:43:24 nixos systemd[1]: Mounting /home/komunix/nfs...
Nov 10 13:43:25 nixos mount[21001]: mount.nfs: access denied by server while mounting synology:/volume2/komunix
Nov 10 13:43:25 nixos systemd[1]: home-komunix-nfs.mount: Mount process exited, code=exited, status=32/n/a
Nov 10 13:43:25 nixos systemd[1]: home-komunix-nfs.mount: Failed with result 'exit-code'.
Nov 10 13:43:25 nixos systemd[1]: Failed to mount /home/komunix/nfs.
Nov 10 13:43:25 nixos systemd[1]: home-komunix-nfs.mount: Consumed 60ms CPU time, 1.5M memory peak, 984B incoming IP traffic, 1.4K outgoing IP traffic.

```

Reply:
> access denied by server while mounting synology:/volume2/komunix

seinget gw acl di synology nya gw grant untuk koneksi dari 100.64.0.0/10 (subnet nya tailscale) ke nfs server. Itu si vm nya jalanin tailscale kan ya?

Reply:
@faultables betul, jalanin tailscale

Reply:
<img width="1135" alt="image" src="https://github.com/user-attachments/assets/cbf00145-82a7-45e9-ba98-317b5127709f">

Reply:
coba lagi brody

Reply:
Nice brody @faultables 

<img width="432" alt="image" src="https://github.com/user-attachments/assets/628ea78b-5b36-43f0-9af8-acfc3d7589b6">

Reply:
It's work for `mounted` but too long for running command `du -sh` (it takes 3min) 🥹

Reply:
No problem BTW! nevermind!